### PR TITLE
MGMT-11182: Allow base level domain

### PR DESCRIPTION
**Pull request by slaviered**
Enable top level domain DNS name to be used for support of local
dns servers.

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [x] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: subsystem/cluster_test.go [hunk 1]
```diff
@@ -909,9 +909,9 @@ var _ = Describe("Validate BaseDNSDomain when creating a cluster", func() {
 	}
 	tests := []DNSTest{
 		{
-			It:            "V2RegisterCluster should throw an error. BaseDNSDomain='example', not a valid DNS structure string",
+			It:            "V2RegisterCluster should not throw an error. BaseDNSDomain='example', valid DNS",
 			BaseDNSDomain: "example",
-			ShouldThrow:   true,
+			ShouldThrow:   false,
```
Comment (mkowalski) [inline subsystem/cluster_test.go:914]: Can you please add a subsystem test with a negative scenario? I.e. it's good to have `domain=example` that will succeed, but I'd also like to see a subsystem test with some domain that will fail, e.g. `basednsdomain=a-.com` and `shouldthrow: true`.

Apart from that looks good, this one test and LGTM

File: subsystem/cluster_test.go [hunk 2]
```diff
@@ -909,9 +909,9 @@ var _ = Describe("Validate BaseDNSDomain when creating a cluster", func() {
...
 		},
 		{
 			It:            "V2RegisterCluster should throw an error. BaseDNSDomain='example.c', Invalid top-level domain name ",
```
Comment (mkowalski) [inline subsystem/cluster_test.go:917]: I am confused here, that's what we discussed offline. Why is `example.c` invalid? Top Level Domain of 1 character is technically a valid one. It is defined in https://www.rfc-editor.org/rfc/rfc952 as `<name>  ::= <let>[*[<let-or-digit-or-hyphen>]<let-or-digit>]`.

Apart from that there is https://datatracker.ietf.org/doc/html/rfc3696#section-2 which states
```
additional rule
that essentially requires that top-level domain names not be all-
numeric
```

So `example.c` is a valid name whereas `example.1` is invalid

File: subsystem/cluster_test.go (hunk 1 above)
Comment (slaviered) [reply subsystem/cluster_test.go:914]: This is a classic example for a code that does not need subsystem at all. The algorithm is tested thoroughly with unit test (as it should be) and there is no need for extra subsystem test that just add a thin layer of calling the algorithm through REST.

Comment (slaviered) [reply subsystem/cluster_test.go:914]: Plus I missed the part of the subsystem but of course fixed it already. "example" should be a legal domain now

File: subsystem/cluster_test.go (hunk 2 above)
Comment (slaviered) [reply subsystem/cluster_test.go:917]: @mkowalski  For some reason the existing validation (which I did not change as it was not part of the PR requirement strictly speaking) requires it to be a 2 or more letter suffix. I Believe it is because we are aiming for a real world DNS address where the last part will be .com or .au). I am not sure we need to relax this limitation. What do you think?

File: subsystem/cluster_test.go (hunk 1 above)
Comment (mkowalski) [reply subsystem/cluster_test.go:914]: Ok, fair enough 👍🏻

File: subsystem/cluster_test.go (hunk 2 above)
Comment (mkowalski) [reply subsystem/cluster_test.go:917]: As I said, real world does allow for a suffix that contains only 1 letter. Sadly I cannot answer why the original regex had the requirement of "minimum 2 characters" but this requirement does not come from the standard (i.e. RFC, anything what is not in RFC is someone's implementation, not "standard"). It may be one of the vendor's implementation (e.g. Microsoft has more strict requirements for DNS names), it may be someone's invented rule, I just can't say why we had it in the first place (just like we don't know why we had a limitation that we need a "dot" so we landed with the issue you are fixing)

Anyway, you are fixing what got initially reported so I'm okay keeping it like this till we get a request to soften it again in the future. The change is too small and too irrelevant to keep it hanging forever :)
